app/scripts/components/common/map/utils.ts

```diff
@@ -123,3 +122,63 @@ export async function requestQuickCache({
   return quickCache.get(key);
 }
 
+type Fn = (...args: any[]) => any;
+
+type ObjResMap<T> = {
+  [K in keyof T]: Res<T[K]>;
+};
+
+type Res<T> = T extends Fn
+  ? T extends DatasetDatumFn<DatasetDatumReturnType>
+    ? DatasetDatumReturnType
+    : never
+  : T extends any[]
+  ? Res<T[number]>[]
+  : T extends object
+  ? ObjResMap<T>
+  : T;
+
```

Comment:
Move to `types.d.ts` ?

Reply:
These types are only ever used for the `resolveConfigFunctions` function declaration and are not needed elsewhere so I think they can live with the declaration. However feel free to move them if you like